src/views/clusteroverview/SettingsTab/ClusterTab/components/ResourceManagementSection/components/KernelSamepageMerging/KernelSamepageMerging.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC, useEffect, useState } from 'react';
+
+import HyperConvergedModel from '@kubevirt-ui/kubevirt-api/console/models/HyperConvergedModel';
+import HelpTextIcon from '@kubevirt-utils/components/HelpTextIcon/HelpTextIcon';
+import useHyperConvergeConfiguration, {
+  HyperConverged,
+} from '@kubevirt-utils/hooks/useHyperConvergeConfiguration';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  PopoverPosition,
+  Split,
+  SplitItem,
+  Switch,
+} from '@patternfly/react-core';
+
+import './KernelSamepageMerging.scss';
+
+const KernelSamepageMerging: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [hyperConverge, hyperLoaded, hyperLoadingError] = useHyperConvergeConfiguration();
+  const [error, setError] = useState<Error>(null);
+
+  useEffect(() => {
+    if (!hyperLoaded || hyperLoadingError) {
+      setError(hyperLoadingError);
+    } else {
+      const ksmConfiguration = hyperConverge?.spec?.configuration?.ksmConfiguration;
+
+      setError(null);
+      setIsEnabled(!!(ksmConfiguration && isEmpty(ksmConfiguration?.nodeLabelSelector))); // Empty nodeLabelSelector will enable KSM on every node.
+    }
+  }, [hyperConverge, hyperLoaded, hyperLoadingError]);
```

Comment:
I don't think this useeffect is needed. you can wait for hyperLoaded and for error in line 75 u can use error || hyperLoadingError

Reply:
Thanks, I'll take care about that!